apps/api/src/database/repositories/agent-repository.ts

```diff
@@ -159,21 +157,20 @@ export async function findAgentCompetitions(
 
     // Check if sorting by computed fields (handled at service layer)
     const isComputedSort =
-      params.sort &&
+      sort &&
       COMPUTED_SORT_FIELDS.some(
-        (field) =>
-          params.sort!.includes(field) || params.sort!.includes(`-${field}`),
+        (field) => sort!.includes(field) || sort!.includes(`-${field}`),
       );
 
     // Only apply database sorting for non-computed fields
-    if (params.sort && !isComputedSort) {
-      query = getSort(query, params.sort, agentCompetitionsOrderByFields);
+    if (sort && !isComputedSort) {
+      query = getSort(query, sort, agentCompetitionsOrderByFields);
     }
 
     // For computed sorting, we'll need to get all results and sort at service layer
     // So we don't apply limit/offset here if sorting by computed fields
     if (!isComputedSort) {
-      query = query.limit(params.limit).offset(params.offset);
+      query = query.limit(limit).offset(offset);
     }
 
     const results = await query;
```

Comment:
Like you mentioned - these changes are just an interim solution and eventually we will replace what's happening here (fetching all agents with `const results = await query` instead of applying pagination at the DB level